src/runtime/virtcontainers/pkg/cgroups/utils.go

```diff
@@ -75,6 +79,68 @@ func IsSystemdCgroup(cgroupPath string) bool {
 	return false
 }
 
+func newProperty(name string, units interface{}) systemdDbus.Property {
+	return systemdDbus.Property{
+		Name:  name,
+		Value: dbus.MakeVariant(units),
+	}
+}
+
+func cgroupHierarchy(path string) (cgroups.Hierarchy, cgroups.Path) {
+	if !IsSystemdCgroup(path) {
+		return cgroups.V1, cgroups.StaticPath(path)
+	} else {
+		slice, unit, _ := getSliceAndUnit(path)
+		cgroupSlicePath, _ := systemd.ExpandSlice(slice)
+		return cgroups.Systemd, cgroups.Slice(cgroupSlicePath, unit)
+	}
+}
+
+func createCgroupsSystemd(slice string, unit string, pid uint32) error {
+	conn, err := systemdDbus.New()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	properties := []systemdDbus.Property{
+		systemdDbus.PropDescription("cgroup " + unit),
+		newProperty("DefaultDependencies", false),
+		newProperty("MemoryAccounting", true),
+		newProperty("CPUAccounting", true),
+		newProperty("BlockIOAccounting", true),
+	}
+
+	// https://github.com/opencontainers/runc/blob/master/docs/systemd.md
+	if strings.HasSuffix(unit, ".scope") {
+		// Otherwise it's a scope, which we put into a Slice=.
+		properties = append(properties, newProperty("Delegate", true))
+		properties = append(properties, systemdDbus.PropSlice(slice))
+		properties = append(properties, systemdDbus.PropPids(pid))
+	} else {
+		// If we create a slice, the parent is defined via a Wants=.
```

Comment:
Since we will detect systemd cgroups driver by the format of the cgroups path passed down by the container engine ([slice]:[prefix]:[name]) where a scope is expected: is there any case where we will need to create a slice and not a scope?

Reply:
You're right, I left as i though it may be used somehow later, I'll remove it for now